interface/ceed-fortran.c

```diff
@@ -631,6 +631,85 @@ void fCeedLobattoQuadrature(int *Q, CeedScalar *qref1d, CeedScalar *qweight1d,
   *err = CeedLobattoQuadrature(*Q, qref1d, qweight1d);
 }
 
+// -----------------------------------------------------------------------------
+// CeedQFunctionContext
+// -----------------------------------------------------------------------------
+static CeedQFunctionContext *CeedQFunctionContext_dict = NULL;
+static int CeedQFunctionContext_count = 0;
+static int CeedQFunctionContext_n = 0;
+static int CeedQFunctionContext_count_max = 0;
+
+#define fCeedQFunctionContextCreate \
+    FORTRAN_NAME(ceedqfunctioncontextcreate,CEEDQFUNCTIONCONTEXTCREATE)
+void fCeedQFunctionContextCreate(int *ceed, int *ctx, int *err) {
+  if (CeedQFunctionContext_count == CeedQFunctionContext_count_max) {
+    CeedQFunctionContext_count_max += CeedQFunctionContext_count_max/2 + 1;
+    CeedRealloc(CeedQFunctionContext_count_max, &CeedQFunctionContext_dict);
+  }
+
+  CeedQFunctionContext *ctx_ =
+    &CeedQFunctionContext_dict[CeedQFunctionContext_count];
+
+  *err = CeedQFunctionContextCreate(Ceed_dict[*ceed], ctx_);
+  if (*err) return;
+  *ctx = CeedQFunctionContext_count++;
+  CeedQFunctionContext_n++;
+}
+
+#define fCeedQFunctionContextSetData \
+    FORTRAN_NAME(ceedqfunctioncontextsetdata,CEEDQFUNCTIONCONTEXTSETDATA)
+void fCeedQFunctionContextSetData(int *ctx, int *memtype, int *copymode,
+                                  CeedInt *n,
+                                  CeedScalar *data, int64_t *offset, int *err) {
+  size_t ctxsize = ((size_t) *n)*sizeof(CeedScalar);
+  *err = CeedQFunctionContextSetData(CeedQFunctionContext_dict[*ctx],
+                                     (CeedMemType)*memtype,
+                                     (CeedCopyMode)*copymode, ctxsize,
+                                     (CeedScalar *)(data + *offset));
```

Comment:
```suggestion
                                     data + *offset);
```
Doesn't need a cast.

Reply:
Is it necessary for `offset` to be part of this interface? I guess it's for symmetry with `CeedVectorSetArray` because it might have been obtained from a different place. But the user can always pass the offset value `data(offset)` in Fortran.

Reply:
I'm not exactly convinced it needs to be in the CeedVector either, but you are correct, I added it for symmetry.